### PR TITLE
[Update] Change buffer type of NDArrayShape and NDArrayStrides to Int

### PR DESCRIPTION
**Pull request by forFudan**
This PR changes the buffer type of `NDArrayShape` and `NDArrayStrides` to `Int`.

## Motivation

Currently, the data buffer of the `NDArrayShape` and `NDArrayStrides` is DType. However, since the indices are of `Int` type, we always pass in and take values as `Int` from these two structs. This means that the data needs to be transferred twice between `Int` and `DType`, one at initialization and one at `__getitem__`.

The only potential advantage of using `DType` is to vectorize strides operations with SIMD. However, the ndim of array is, in most cases, small (less than 4). The gain of vectorization is marginal. Moreover, vectorizing using SIMD means that we need to first transfer list of indices also to SIMD, this means extra time to be consumed.

To summarize, it is better to directly use `Int` type than `DType`.

### CODE REVIEW

**Comment by forFudan** [comment]
@shivasankarka would you try the "squash and merge" and give a title and information to the merge?

**Review by shivasankarka — APPROVED** [review]
